src/components/PendingTxsSnackbar/store.js

```diff
@@ -61,27 +61,49 @@ export default class PendingTxsSnackbarStore {
 
   @action
   queryPendingTransactions = async () => {
+    // Address is required for the request filters
+    if (isEmpty(this.app.wallet.addresses)) {
+      this.reset();
+      return;
+    }
+
     try {
-      const filters = [{ status: TransactionStatus.PENDING }];
+      const filters = reduce(this.app.wallet.addresses, (result, obj) => {
+        result.push({ status: TransactionStatus.PENDING, senderAddress: obj.address });
+        return result;
+      }, []);
+
       const result = await queryAllTransactions(filters);
-      const txs = _.map(result, (tx) => new Transaction(tx));
+      const txs = map(result, (tx) => new Transaction(tx));
+      const {
+        APPROVE_CREATE_EVENT,
+        CREATE_EVENT,
+        BET,
+        APPROVE_SET_RESULT,
+        SET_RESULT,
+        APPROVE_VOTE,
+        VOTE,
+        FINALIZE_RESULT,
+        WITHDRAW,
+        WITHDRAW_ESCROW,
+        TRANSFER,
+        RESET_APPROVE,
+      } = TransactionType;
 
       runInAction(() => {
         this.count = txs.length;
-        this.pendingCreateEvents = _.filter(
+        this.pendingApproves = filter(
           txs,
-          (tx) => tx.type === TransactionType.APPROVE_CREATE_EVENT || tx.type === TransactionType.CREATE_EVENT
+          (tx) => tx.type === APPROVE_CREATE_EVENT || tx.type === APPROVE_SET_RESULT || tx.type === APPROVE_VOTE,
         );
-        this.pendingBets = _.filter(txs, { type: TransactionType.BET });
-        this.pendingSetResults = _.filter(txs, (tx) =>
-          tx.type === TransactionType.APPROVE_SET_RESULT || tx.type === TransactionType.SET_RESULT);
-        this.pendingVotes = _.filter(txs, (tx) =>
-          tx.type === TransactionType.APPROVE_VOTE || tx.type === TransactionType.VOTE);
-        this.pendingFinalizeResults = _.filter(txs, { type: TransactionType.FINALIZE_RESULT });
-        this.pendingWithdraws = _.filter(txs, (tx) =>
-          tx.type === TransactionType.WITHDRAW || tx.type === TransactionType.WITHDRAW_ESCROW);
-        this.pendingTransfers = _.filter(txs, { type: TransactionType.TRANSFER });
-        this.pendingResetApproves = _.filter(txs, { type: TransactionType.RESET_APPROVE });
+        this.pendingCreateEvents = filter(txs, { type: CREATE_EVENT });
+        this.pendingBets = filter(txs, { type: BET });
+        this.pendingSetResults = filter(txs, { type: SET_RESULT });
+        this.pendingVotes = filter(txs, { type: VOTE });
+        this.pendingFinalizeResults = filter(txs, { type: FINALIZE_RESULT });
+        this.pendingWithdraws = filter(txs, (tx) => tx.type === WITHDRAW || tx.type === WITHDRAW_ESCROW);
+        this.pendingTransfers = filter(txs, { type: TRANSFER });
+        this.pendingResetApproves = filter(txs, { type: RESET_APPROVE });
```

Comment:
So those are actually arrays of pending events, in its `index.js` line 72, we use `amounts` to represent those arrays, I think amounts is misleading

Reply:
sure, renamed to `txs`